src/StatCache.cc

```diff
@@ -0,0 +1,65 @@
+/*
+ *  OpenSCAD (www.openscad.org)
+ *  Copyright (C) 2009-2011 Clifford Wolf <clifford@clifford.at> and
+ *                          Marius Kintel <marius@kintel.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include "StatCache.h"
+#include "printutils.h"
+
+#include <sys/stat.h>
+#include <string>
+#include <unordered_map>
+
+typedef std::unordered_map<std::string, struct stat> StatMap;
+
+static StatMap statMap;

```

Comment:
The life of the cache is just while each window runs includesChanged() and handleDependencies(), so having a cache per window would not make any difference. As it stands now each window will scan just the files it uses. If those files overlap with the set from another window then they will get scanned twice but at slightly different times.

To get a further speed up with multiple Windows then the cached results need to be reused. That means it has to be a global cache but it should be cleared less often. I.e. about once per 200ms as that is how often each window checks for changes.

I could remove the clear function and put a timestamp for each entry in the cache. When it is more than 200ms old I could consider it stale and re-run the stat. That way each unique file would be scanned every 200ms and the results shared between windows.

BTW, why has the Travis build got stuck at the osx build?